### PR TITLE
completed except three and four not passing test see notes

### PR DESCRIPTION
**Pull request by elysemontano**
I spent hours trying to get 3 and 4 to work, and I cannot seem to get them to pass the test.  Tried many refactors, but no luck.  Spent a lot of time researching options on how to approach them.  

I also don't know if the react challenge was successful.  I do not see the emoji's on the page, however I did find a few bugs and commented what I changed.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -17,6 +17,30 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
+// will describe when divByThree arguement is given
+describe("when divByThree is called", () => {
```
Comment (minachuong) [inline code-challenges.test.js:21]: 💯 awesome tests

File: code-challenges.test.js [hunk 2]
```diff
@@ -27,59 +51,138 @@ var num3 = -7
...
 
-
-
-
+//create a funtion that takes a number as an arguement
+//check number to see if it is divisible by 3 using modulo
+//use if else statement to determine possible outcomes
+  //if given number is divisible by three -- return is divisible
+  //else return is not divisible
+const divByThree = (num) => {
```
Comment (minachuong) [inline code-challenges.test.js:61]: yup! if you're looking for a challenge, consider finding edge cases with testing and providing logic to support it in your function

File: code-challenges.test.js [hunk 3]
```diff
@@ -27,59 +51,138 @@ var num3 = -7
...
-
 // b) Create the function that makes the test pass.
 
+//create a function with an array as an arguement
+//set a variable that will map the arrays
+//the return of the map will have toUpperCase for only the index of 0 and will join back with the vaule with substring.
 
+const arrayCaps = (array) => {
+  let newArrayToCaps = array.map(value => {
+    return value[0].toUpperCase() + value.substring(1)
```
Comment (minachuong) [inline code-challenges.test.js:106]: 💯

File: code-challenges.test.js [hunk 4]
```diff
@@ -27,59 +51,138 @@ var num3 = -7
...
 // Expected output: [-8, 0, 8, 46, 59, 107]
 
-var mixedDataArray1 = [3, "yo!", 94, -9, false, 0, 18, "hola!"]
+var mixedDataArray2 = [3, "yo!", 94, -9, false, 0, 18, "hola!"]
 // Expected output: [-9, 0, 3, 18, 94]
 
 
+// b) Create the function that makes the test pass.
 
+//make a function that takes an array as an arguement with multiple data types
```
Comment (minachuong) [inline code-challenges.test.js:142]: 👍 great pseudocoding

File: code-challenges.test.js [hunk 5]
```diff
@@ -27,59 +51,138 @@ var num3 = -7
...
-// b) Create the function that makes the test pass.
-
+//return values as a new array
 
+const numOnly = (array) => {
+  let numArray = array.filter (value => {
+    if (typeof value === 'number')
+      return value
+      })
+    for(let i = 0; i < numArray.length; i++)
```
Comment (minachuong) [inline code-challenges.test.js:154]: reconsider the placement of this for-loop because it's happening within an iteration (`.filter()`) already. It would read a bit clearer that you first complete the filtering then run the for-loop

File: code-challenges.test.js [hunk 6]
```diff
@@ -92,3 +195,35 @@ var vowelTester3 = "challenge"
...
+//   return firstVowelArray
+// }
+//
+// console.log(vowelFinder(vowelTester1));
+// console.log(vowelFinder(vowelTester2));
+// console.log(vowelFinder(vowelTester3));
+
+const vowelFinder = (string) => {
+  let vowelArray = string.split("")
+  for(let i = 0; i < vowelArray.length; i++){
```
Comment (minachuong) [inline code-challenges.test.js:221]: consider using `.forEach()` instead of a for-loop. You can still have access to an index by doing something like 
```
vowelArray.forEach((letter, index) => { ... })
```

File: react-challenge.html [hunk 7]
```diff
@@ -23,36 +23,40 @@
...
-            currentMood: ""
+            currentMood: [""]
           }
         }
 
         handleMoodOptions = () => {
-          let randomIndex = Math.floor(Math.random() * this.state.moods.length )
-          this.setState({ currentMood: moods[randomIndex] })
+          let randomIndex = Math.floor(Math.random() *  this.state.moods.length )
+          this.setState({ currentMood: this.state.moods[randomIndex] })
```
Comment (minachuong) [inline react-challenge.html:39]: 💯 yes, moods lives in state and therefore we need to reference `this.state.`

File: react-challenge.html [hunk 8]
```diff
@@ -23,36 +23,40 @@
...
         }
 
         render(){
+          let { currentMood } = this.state
```
Comment (minachuong) [inline react-challenge.html:43]: very cool refactor using destructuring!

File: react-challenge.html [hunk 9]
```diff
@@ -23,36 +23,40 @@
...
           return(
 
+              //the <></> were missing to nest multiple elements
+              //was missing this. for the onClick
+              <>
```
Comment (minachuong) [inline react-challenge.html:48]: yes! every component that has more than one HTML element needs to be encased in a react fragment

File: react-challenge.html [hunk 10]
```diff
@@ -23,36 +23,40 @@
...
 
               <h3>Mood Options:</h3>
 
               <div style={{ fontSize: "40px", height: "50px"}}>
-                { this.state.currentMood }
+                { this.currentMood }
               </div>
-
-              <button onClick={ handleMoodOptions }>
+              <button onClick={ this.handleMoodOptions }>
```
Comment (minachuong) [inline react-challenge.html:56]: yes! because we're referencing a class method we need to properly access it through the keyword `this`

File: react-challenge.html [hunk 11]
```diff
@@ -23,36 +23,40 @@
...
-
+//added brackets to quotation marks for current mood
+//added this.state.moods to this.setState section.
       class App extends React.Component{
         constructor(props){
           super(props)
           this.state = {
             moods: ["😃", "😴", "😍", "🥺", "😡", "🤓", "🤪"],
-            currentMood: ""
+            currentMood: [""]
```
Comment (minachuong) [inline react-challenge.html:33]: Looks like this was changed from string to array of string, which is fine, it just means all the places that need to access currentMood will need to access the array values by using index

**Review by minachuong — APPROVED** [review]
Keep up the work on pseudocode! Good test assertions. Avoid using for-loops because forEach() is easier to read and you can still leverage index value on an item using the second default parameter in the function you pass into it. Example in comments.
